Share/Share.Plugin.iOSUnified/ShareImplementation.cs

```diff
@@ -113,35 +132,50 @@ static ShareImplementation()
         /// <param name="excludedActivityTypes">UIActivityType to exclude</param>
-        /// <returns>awaitable Task</returns>
-        public async Task ShareLink(string url, string message = null, string title = null, params NSString[] excludedActivityTypes)
+        /// <returns>True if the operation was successful, false otherwise</returns>
+        [Obsolete("Use Share(ShareMessage, ShareOptions)")]
+        public Task<bool> ShareLink(string url, string message = null, string title = null, params NSString[] excludedActivityTypes)
         {
-            await ShareInternal(title, message, url, excludedActivityTypes);
+            var shareMessage = new ShareMessage();
+            shareMessage.Title = title;
+            shareMessage.Text = message;
+            shareMessage.Url = url;
+
+            var shareOptions = new ShareOptions();
+            shareOptions.ExcludedUIActivityTypes = excludedActivityTypes?.Select(x => (string)x).ToArray();
+
+            return Share(shareMessage, shareOptions);
         }
 
         /// <summary>
-        /// Share data with compatible services
+        /// Share a message with compatible services
         /// </summary>
-        /// <param name="title">Title to share</param>
         /// <param name="message">Message to share</param>
-        /// <param name="url">Link to share</param>
-        /// <param name="excludedActivityTypes">UIActivityType to excluded</param>
-        /// <returns>awaitable Task</returns>
-        async Task ShareInternal(string title, string message, string url, NSString[] excludedActivityTypes)
+        /// <param name="options">Platform specific options</param>
+        /// <returns>True if the operation was successful, false otherwise</returns>
+        public async Task<bool> Share(ShareMessage message, ShareOptions options = null)
         {
+            if (message == null)
+                throw new ArgumentNullException(nameof(message));
+
             try
             {
+                // create activity items
                 var items = new List<NSObject>();
-                if (message != null)
-                    items.Add(new ShareActivityItemSource(new NSString(message), title));
-                if (url != null)
-                    items.Add(new ShareActivityItemSource(NSUrl.FromString(url), title));
+                if (message.Text != null)
+                    items.Add(new ShareActivityItemSource((NSString)message.Text, message.Title));
+                if (message.Url != null)
+                    items.Add(new ShareActivityItemSource(NSUrl.FromString(message.Url), message.Title));
 
+                // create activity controller
                 var activityController = new UIActivityViewController(items.ToArray(), null);
 
+                // set excluded activity types
+                var excludedActivityTypes = options?.ExcludedUIActivityTypes?.Select(x => (NSString)x).ToArray() ?? ExcludedUIActivityTypes?.ToArray();
```

Comment:
Use options.ExcludedUIActivityTypes if not null, ShareImplementation.ExcludedUIActivityTypes otherwise
This way the dev can set options.ExcludedUIActivityTypes to null (this is the default) to use the statically configured list, or empty array to override the static list